### PR TITLE
o/devicestate: unset recovery_system when done seeding

### PR DESCRIPTION
**Pull request by anonymouse64**
Also make a drive-by tweak to a variable declaration in firstboot.

### CODE REVIEW

File: overlord/devicestate/handlers.go [hunk 1]
```diff
@@ -99,9 +99,13 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	if m.preseed {
 		return fmt.Errorf("internal error: mark-seeded task not expected in pre-seeding mode")
 	}
-	// TODO:UC20: update "modeenv" and remove "recovery_system" from
-	// it because this information is only needed for the initial
-	// seeding.
+
+	// unset recovery_system because that is only needed during install mode
+	m.modeEnv.RecoverySystem = ""
+	err := m.modeEnv.Write("")
```
Comment (pedronis) [inline overlord/devicestate/handlers.go]: we can't do this unconditionally, we don't want this file to show up on UC16/18. Also I think we want to keep some trail in state about good recovery systems we managed to seed but that part is not a blocker for this.

Comment (anonymouse64) [reply overlord/devicestate/handlers.go]: ah yes you're right... sorry I forgot there's these things not known as uc20 we happen to run snapd on sometimes

Comment (pedronis) [reply overlord/devicestate/handlers.go]: maybe grab a DeviceCtx (see handlers_install.go) and check HasModeenv and OperatingMode is "run" ?

File: overlord/devicestate/firstboot20_test.go [hunk 2]
```diff
@@ -118,12 +117,14 @@ volumes:
...
+	m := boot.Modeenv{
 		Mode:           "run",
 		RecoverySystem: "20191018",
 		Base:           "core20_1.snap",
-	})
-	defer r()
+	}
+	err := m.Write("")
+	c.Assert(err, IsNil)
+	defer os.Remove(dirs.SnapModeenvFileUnder(dirs.GlobalRootDir))
```
Comment (mvo5) [inline overlord/devicestate/firstboot20_test.go]: (nitpick) I don't think we need this, the firstBootBaseTest runs dirs.SetRootDir(tmpdir) and that tmpdir is created via c.MkDir() so it's automatically cleaned up.

**Pull request merged**